test/suites-ostree/staged-failed-upgrade.robot

```diff
@@ -0,0 +1,99 @@
+*** Settings ***
+Documentation       Tests related to upgrading MicroShift
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+${TARGET_REF}       ${EMPTY}
+
+
+*** Test Cases ***
+First Failed Staged Upgrade Should Restore Backup
+    [Documentation]    Staged deployment is unhealthy on first boot. After auto-greenboot-reboot MicroShift
+    ...    should restore rollback deployment's backup to reattempt the upgrade from healthy data.
+
+    Wait For Healthy System
+
+    ${initial_backup}=    Get Future Backup Name For Current Boot
+    Create Backup With Marker    ${initial_backup}
+
+    TestAgent.Add Action For Next Deployment    1    fail_greenboot
+
+    # We don't expect a rollback here since after the first failure
+    # things should continue as desired. This test validates the
+    # restore function is working when the first boot is unhealthy.
+    Deploy Commit Not Expecting A Rollback    ${TARGET_REF}    ${TRUE}
+
+    Wait For Healthy System
+
+    Validate Backup Is Restored    ${initial_backup}
+    Expected Boot Count    3
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Should Not Be Empty    ${TARGET_REF}    TARGET_REF variable is required
+    Login MicroShift Host
+    Setup Kubeconfig
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Create Backup With Marker
+    [Documentation]    Stops MicroShift and creates manual backup that
+    ...    masquerades as automated one and creates marker file for validating restore.
+    [Arguments]    ${backup_name}
+    Systemctl    stop    microshift.service
+
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    microshift backup --name "${backup_name}"
+    ...    sudo=True    return_stderr=True    return_rc=True
+
+    # create a marker that we expect to show up in data directory after restore
+    ${mark_stdout}    ${mark_stderr}    ${mark_rc}=    Execute Command
+    ...    touch ${BACKUP_STORAGE}/${backup_name}/marker
+    ...    sudo=True    return_stderr=True    return_rc=True
+
+    Should Be Equal As Integers    0    ${rc}
```

Comment:
I would check this before creating the marker - what if making a backup fails and you try to create marker anyway? Not a big deal, but might be confusing when debuggin

Reply:
Good call out, yeah this just happened now where it failed due to the backup creation failed due to the change in the subcommand for backup. Updating to fail early here. Thanks!